google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -599,6 +598,25 @@ def validate_snapshot_args! strong: nil, timestamp: nil, staleness: nil
                "Can only provide one of the following arguments: " \
                "(strong, timestamp, staleness)"
         end
+
+        ##
+        # Retrieves the delay value from Google::Cloud::AbortedError
+        def delay_from_aborted err
+          return 0 if err.nil?
+          if err.respond_to?(:metadata) && err.metadata["retryDelay"]
+            # a correct metadata will look like this:
+            # "{\"retryDelay\":{\"seconds\":60}}"
+            seconds = err.metadata["retryDelay"]["seconds"].to_i
+            nanos = err.metadata["retryDelay"]["nanos"].to_i
+            return seconds if nanos.zero?
+            return seconds + (nanos / 1000000000.0)
+          end
+          # No metadata? Try the inner error
+          delay_from_aborted(err.cause)
+        rescue
+          # Any error gets the default delay, which is 0
+          return 0
```

Comment:
No, GAPIC does not handle errors during streaming responses. The GRPC errors are passed through the Enumerator object that responses are pulled from, and the GAPIC layer isn't involved. The GAPIC layer will only handles errors on the call that instigates the streaming request and creates the Enumerator object.